lightning/src/chain/chainmonitor.rs

```diff
@@ -431,9 +656,33 @@ where C::Target: chain::Filter,
 	}
 
 	fn release_pending_monitor_events(&self) -> Vec<MonitorEvent> {
-		let mut pending_monitor_events = Vec::new();
+		let mut pending_monitor_events = self.pending_monitor_events.lock().unwrap().split_off(0);
 		for monitor_state in self.monitors.read().unwrap().values() {
-			pending_monitor_events.append(&mut monitor_state.monitor.get_and_clear_pending_monitor_events());
+			let is_pending_monitor_update = monitor_state.has_pending_chainsync_updates(&monitor_state.pending_monitor_updates.lock().unwrap());
+			if is_pending_monitor_update &&
+					monitor_state.last_chain_persist_height.load(Ordering::Acquire) + LATENCY_GRACE_PERIOD_BLOCKS as usize
+						> self.highest_chain_height.load(Ordering::Acquire)
```

Comment:
Can there be an edge case around reorgs here where `highest_chain_height` is reduced while blocks are disconnected, `release_pending_monitor_events` is called, and then the new blocks are connected?

Reply:
Hmm, yea, was thinking it wasn't worth worrying about, but its easy enough to just `fetch_max`, so I did that.